azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -1464,6 +1458,15 @@ public FlowRunnerEventListener() {
       metaData.put("startTime", String.valueOf(flow.getStartTime()));
       metaData.put("submitTime", String.valueOf(flow.getSubmitTime()));
 
+      // project upload info
+      final ProjectFileHandler handler = flowRunner.projectFileHandler;
+      if (handler != null) {
```

Comment:
Can the file handler be null and shouldn't we throw an exception in that case ?

Reply:
This is to keep the existing unit tests happy as the info comes from db where it does not exist. If there is a way to circumvent in tests, I would like to know.
Same reason why there is no unit test for this change.

Reply:
How many tests fail without the conditional? I think we should fix the tests

Reply:
flowEventMetadata fails as it tries to generate metadata. The FlowRunner object has projectFileHandler set to NULL as there is no db to pull it from. I dont know how to make it happen for test that is why I kept that null check.